rdmd.d

```diff
@@ -59,6 +59,15 @@ version(unittest) {} else
 int main(string[] args)
 {
     //writeln("Invoked with: ", args);
+    string binPath = thisExePath();
+    if (binPath.exists && binPath.isFile && filenameCmp(baseName(binPath),
+                                                        baseName(args[0])) == 0)
```

Comment:
I don't think this is the right approach. Regardless, the last check may fail on Windows or other systems with case-insensitive filesystems.